pkg/controller/mutators/stats_reporter_test.go

```diff
@@ -72,23 +72,121 @@ func TestReportMutatorIngestionRequest(t *testing.T) {
 		t.Errorf("got tag %q max %v, want %v", mutatorIngestionDurationMetricName, durationValue.Max, wantMaxIngestionDuration)
 	}
 
-	verifyTags(t, wantTags, row.Tags)
+	verifyTags(t, wantTags, rows[0].Tags)
 }
 
-func checkData(name string, rowLength int) (*view.Row, error) {
-	row, err := view.RetrieveData(name)
+func TestReportMutatorsStatus(t *testing.T) {
+	r := NewStatsReporter()
+
+	activeMutators := 5
+	errorMutators := 1
+	if err := r.ReportMutatorsStatus(MutatorStatusActive, activeMutators); err != nil {
+		t.Errorf("ReportMutatorsStatus error %v", err)
+	}
+	if err := r.ReportMutatorsStatus(MutatorStatusError, errorMutators); err != nil {
+		t.Errorf("ReportMutatorsStatus error %v", err)
+	}
+
+	rows, err := checkData(mutatorsMetricName, 2)
 	if err != nil {
-		return nil, fmt.Errorf("got RetrieveData error %v from %v, want nil", err, name)
+		t.Error(err)
+	}
+
+	activeRow, err := getRow(rows, string(MutatorStatusActive))
+	if err != nil {
+		t.Error(err)
+	}
+
+	lastValueData, ok := activeRow.Data.(*view.LastValueData)
+	if !ok {
+		t.Fatalf("wanted active status row of type LastValueData. got: %v", activeRow.Data)
+	}
+
+	if lastValueData.Value != float64(activeMutators) {
+		t.Errorf("wanted status %q to have value %v, got: %v", MutatorStatusActive, activeMutators, lastValueData.Value)
+	}
+
+	errorRow, err := getRow(rows, string(MutatorStatusError))
+	if err != nil {
+		t.Error(err)
+	}
+
+	lastValueData, ok = errorRow.Data.(*view.LastValueData)
+	if !ok {
+		t.Fatalf("wanted error status row of type LastValueData. got: %v", activeRow.Data)
```

Comment:
Use %T for reporting the desired type. That keeps the generated error message from drifting from the actually-desired type. Also - use %T for `activeRow.Data` so the test message includes the actual type.

Reply:
nice, will change

Reply:
Going to hold off on this change until after we resolve [this other thread](https://github.com/open-policy-agent/gatekeeper/pull/1714#discussion_r761536594), as it may involve deleting code I'd be updating for this comment.

Reply:
Done